pyam/core.py

```diff
@@ -951,15 +951,17 @@ def _exclude_on_fail(self, df):
         logger().info('{} non-valid scenario{} will be excluded'
                       .format(len(idx), '' if len(idx) == 1 else 's'))
 
-    def filter(self, keep=True, inplace=False, **kwargs):
+    def filter(self, keep=True, inplace=False, warn_if_empty=True, **kwargs):
```

Comment:
not sure, but shouldn't it be configurable via setting log level?

Reply:
the point is that in some particular cases such as the one where the new option is used, there should *never*  be a log message

Reply:
I think it can be a bit cleaner to explicitly catch expected warnings a la [`catch_warnings`](https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings)